packages/components/src/divider/component.tsx

```diff
@@ -17,11 +17,46 @@ import {
 import { DividerView } from './styles';
 import type { Props } from './types';
 
+function useDeprecatedProps( {
+	marginBottom,
+	marginTop,
+	marginStart,
+	marginEnd,
+	...otherProps
+}: WordPressComponentProps< Props, 'hr', false > ) {
+	const propsToReturn: WordPressComponentProps< Props, 'hr', false > = {
+		...otherProps,
+	};
+
+	// Transform deprecated `marginTop` prop into `marginStart`.
+	let computedMarginStart = marginStart;
+	if ( marginTop ) {
+		computedMarginStart ??= marginTop;
+	}
+	if ( typeof computedMarginStart !== 'undefined' ) {
+		propsToReturn.marginStart = computedMarginStart;
+	}
```

Comment:
I initially came up with this pattern on the `Card` component in order to avoid overriding any potential props passed via Context: https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/card/card/hook.js#L35

Probably safer to keep using this pattern to future-proof `Divider`, what do you think?  (in that case, I will definitely add some comments as you suggested)

Reply:
Now that I'm thinking about it, this is an experimental component — and therefore another option could be to just remove support for `marginTop` and `marginBottom` (avoiding `useDeprecatedProps` altogether)

Reply:
> avoid overriding any potential props passed via Context

Ah that's good to know! 😮 Do you think it's worth addressing this in the `useContextSystem` function itself so it's smarter about undefined properties? It's pretty easy to mess up if you don't know or forget.

> another option could be to just remove support

That's perfectly reasonable!

Reply:
> Ah that's good to know! 😮 Do you think it's worth addressing this in the useContextSystem function itself so it's smarter about undefined properties? It's pretty easy to mess up if you don't know or forget.

I'm not sure, but it's definitely something to look into

> That's perfectly reasonable!

Reply:
Done in https://github.com/WordPress/gutenberg/pull/36579/commits/0f6000120dde2cb05dc6417435b977d8206ea1a5